semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,456 @@
+    EnkfNode,
+)
+from res.enkf import (
+    ESUpdate,
+    ErtRunContext,
+)
+from res.enkf.export import (
+    GenKwCollector,
+    MisfitCollector,
+)
+
+from ert_shared.plugins.plugin_manager import hook_implementation
+from ert_shared.libres_facade import LibresFacade
+
+
+DESCRIPTION = """
+Purpose is to calculate (ERT workflow)
+the degree of update (KS matrix)
+of the different parameters with regards to the different
+observations. Create csv files which can be used in
+Webviz/Jupyternotebook/dash to perform analysis of the AHM.
+requires an ert config file as input
+optionally a name can be set for the target case used for
+update (ahm) calculation
+"""
+
+
+class AhmAnalysisJob(SemeioScript):  # pylint: disable=too-few-public-methods
+    """ Define ERT workflow to evaluate change of parameters for each observation"""
+
+    def run(self, target_name="analysis_case", prior_name=None, group_by="data_key"):
+        """Perform analysis of parameters change per obs group
+        prior to posterior of ahm"""
+        ert = self.ert()
+        facade = LibresFacade(self.ert())
+
+        obs_keys = [
+            facade.get_observation_key(nr)
+            for nr, _ in enumerate(facade.get_observations())
+        ]
+        key_map = _group_observations(facade, obs_keys, group_by)
+
+        prior_name, target_name = check_names(
+            facade.get_current_case_name(),
+            prior_name,
+            target_name,
+        )
+        # Get the prior scalar parameter distributions
+        prior_data = check_inputs(
+            MisfitCollector.loadAllMisfitData(ert, prior_name),
+            GenKwCollector.loadAllGenKwData(ert, prior_name),
+        )
+
+        # create dataframe with observations vectors (1 by 1 obs and also all_obs)
+        combinations = make_obs_groups(key_map)
+        # setup dataframe for calculated data
+        ks_data, active_obs, misfitval = initialize_emptydf()
+
+        field_parameters = sorted(
+            ert.ensembleConfig().getKeylistFromImplType(ErtImplType.FIELD)
+        )
+        scalar_parameters = sorted(
+            ert.ensembleConfig().getKeylistFromImplType(ErtImplType.GEN_KW)
+        )
+        # identify the set of actual parameters that was updated for now just go
+        # through scalar parameters but in future if easier access to field parameter
+        # updates should also include field parameters
+        dkeysf = get_updated_parameters(prior_data, scalar_parameters)
+
+        # loop over keys and calculate the KS metric,
+        # conditioning one parameter at the time.
+        field_output = {}
+        for group_name, obs_group in combinations.items():
+            print("Processing:", group_name)
+
+            #  Use localization to evaluate change of parameters for each observation
+            update_log_path = create_path(
+                ert.getModelConfig().getRunpathAsString(), target_name
+            ).replace("scalar_", "update_log")
```

Comment:
This doesn't change in the for loop, we can put it outside

Reply:
Or, I don't think I understand this.. `create_path` simply returns a tmpdir? And replace wouldn't do any good? And you wouldn't get a file named `update_log` either then?

Reply:
That was a change I made, as the `create_path` function originally did something else, and I forgot to change the inputs and the `replace`. I would suggest doing this:

```
            #  Use localization to evaluate change of parameters for each observation
            with tempfile.TemporaryDirectory() as update_log_path:
                _run_ministep(
                    ert,
                    facade,
                    obs_group,
                    field_parameters + scalar_parameters,
                    prior_name,
                    target_name,
                    update_log_path,
                )
                # Get the inactive vs total observation info
                df_update_log = make_update_log_df(update_log_path, group_name)
```
then we get automatic clean up of the temporary folder.

Reply:
done